test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -162,24 +163,31 @@ database_operation::insert_operation(thread_context *tc)
         auto &cc = ccv[counter];
         while (tc->transaction.active() && tc->running()) {
             /* Insert a key value pair. */
-            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
-                committed = false;
-                break;
+            bool rollback_required = tc->insert(cc.cursor, cc.coll.id, start_key + added_count);
```

Comment:
To avoid the repetition of checks on `rollback_required`, I suggest the following:
```
            bool rollback_required = false; //Could be declared before the loop, but set to false at the start of every iteration 
            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
                added_count++;
                if (tc->transaction.can_commit()) {
                    if(!tc->transaction.commit())
                        /*
                         * We need to inform the database model that we've added these keys as some
                         * other thread may rely on the key_count data. Only do so if we
                         * successfully committed.
                         */
                        cc.coll.increase_key_count(added_count);
                    else
                        rollback_required = false;
                }
            } else
                rollback_required = true;
```

Reply:
This seems a bit less clean to me?

Reply:
You can leave it then, it was only a suggestion !